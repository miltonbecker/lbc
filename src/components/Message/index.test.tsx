import React from 'react';
import renderer from 'react-test-renderer';
import { Message } from './';

describe('Message', () => {
  it('Renders confidential messages correctly', () => {
    const tree = renderer
      .create(
        <Message
          key="1"
          message={{ id: '1', author: 'Jean', text: 'Test message.', confidential: true }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Renders non confidential messages correctly', () => {
    const tree = renderer
      .create(
        <Message
          key="1"
          message={{ id: '1', author: 'Jean-Pierre', text: 'Test public message.', confidential: false }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
