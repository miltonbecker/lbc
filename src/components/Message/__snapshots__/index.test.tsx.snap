// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Renders confidential messages correctly 1`] = `
<div
  className="message confidential"
>
  <p
    className="author"
  >
    Jean
  </p>
  <p>
    Test message.
  </p>
</div>
`;

exports[`Message Renders non confidential messages correctly 1`] = `
<div
  className="message"
>
  <p
    className="author"
  >
    Jean-Pierre
  </p>
  <p>
    Test public message.
  </p>
</div>
`;
